Add Silero Speech-To-Text models

snakers4 opened the pull request:
Please kindly review our Speech-To-Text models

snakers4 commented:
Maybe I did something wrong, but I cannot see our model's page [here](https://deploy-preview-153--pytorch-hub-preview.netlify.app/hub/research-models)

snakers4 commented:
Hi, @ailzhang @bertmaher @wconstab could you please help with assigning a correct person to review the submission?
Thanks!

ailzhang reviewed (changes requested):
@snakers4 
Thanks for contributing!
If you move your `md` file out of `docs` you should be able to see it on the webpage. 
Also once you do that, code snippet in the the markdown will be executed in the CI so you probably also need a working example there (e.g. `test_files`)

snakers4 commented:
Hi,

> If you move your md file out of docs you should be able to see it on the webpage.

Moved it to the root folder

> Also once you do that, code snippet in the the markdown will be executed in the CI so you probably also need a working example there (e.g. test_files)

Added a small validation dataset download so that it works end-to-end, tested it locally

snakers4 commented:
fixed the imports and path issues
hopefully it will be fine this time

snakers4 commented:
@ailzhang 
```
Running pytorch example in snakers4_silero-models_stt.md
Traceback (most recent call last):
  File "temp.py", line 3, in <module>
    import torchaudio
ModuleNotFoundError: No module named 'torchaudio'
```
Looks like the build failed because there is no torchaudio in the build environment
It seems that I have specified torchaudio in the dependencies list [here](https://github.com/snakers4/silero-models/blob/master/hubconf.py#L1)
Can you please help with this?

snakers4 commented:
@ailzhang 

looks like all is fixed
please take a look

soumith commented on snakers4_silero-models_stt.md:
```diff
@@ -0,0 +1,77 @@
+# see https://github.com/snakers4/silero-models for utils and more examples
+
+device = torch.device('cpu')  # gpu also works, but our models are fast enough for CPU
+model, decoder, utils = torch.hub.load(github='snakers4/silero-models',
```
would this require omegaconf and torchaudio  to be installed?

if so, you should add a cell like in https://github.com/pytorch/hub/blob/master/nvidia_deeplearningexamples_waveglow.md#example
which pip installs these extra packages.

that will make the thing instantly run in Google Colab, and is really valuable!

snakers4 replied:
Hi,

> would this require omegaconf and torchaudio to be installed?

Yeah, this would 
That is why I needed to include them in your CI environment above
Actually in colab I also needed to install soundfile, as we are using it as a backend for TorchAudio

> if so, you should add a cell like in
> that will make the thing instantly run in Google Colab, and is really valuable!

yeah, this totally makes sense
by the way, you can see a more extended colab version [here](https://colab.research.google.com/github/snakers4/silero-models/blob/master/examples.ipynb)

I will add this shortly

soumith commented on snakers4_silero-models_stt.md:
```diff
@@ -0,0 +1,77 @@
+ read_audio,
+ prepare_model_input) = utils  # see function signature for details
+
+torch.hub.download_url_to_file('http://www.openslr.org/resources/83/midlands_english_female.zip',
```
instead of a zip file, extracting it and then running a batch of wav files through, it would be much nicer and more illustrative of downloading a single wav file and processing it through.

like:

```
torch.hub.download_url_to_file('some download url for speech.wav', dst='speech.wav)

input = prepare_model_input(glob('midlands_english_female/*.wav'), device=device)

output = model(input)
```

snakers4 replied:
one of reasons why I am downloading whole validation dataset and running a batch is to demonstrate the models are actually quite fast on CPU as well as on GPU

but I see your point 
I will add a default example with one file, but I would nevertheless keep the utils
because for a first time user batching may be an issue, so I would like to solve that

I will add the changes shortly

soumith reviewed (commented):
great great addition.

I have one important comment around dependencies (so that the hub example works in Google Colab) and another comment which is a suggestion.
Please take a look

soumith commented:
thanks for the great contribution. it should go live sometime today, maybe in a couple of hours.

snakers4 commented:
checked [here](https://pytorch.org/hub/research-models) I guess it is not there yet, looking forward to the release!

on a side note, as a small self-funded team we feel extremely proud to have made something worthy of including in this hub
speech has been asking for some care for quite some time

soumith commented:
apparently we've moved to a once-per-day update of the site now. it should be live by tomorrow morning.

snakers4 commented:
it is live now
